internal/gateway/statistics/statistics.go

```diff
@@ -0,0 +1,69 @@
+package statistics
+
+import (
+	"fmt"
+	"os/exec"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+type Counters struct {
+	ChunksAdded          int64  `json:"Publish.n_chunks_added,string"`
+	ChunksDuplicated     int64  `json:"Publish.n_chunks_duplicated,string"`
+	CatalogsAdded        int64  `json:"Publish.n_catalogs_added,string"`
+	UploadedBytes        int64  `json:"Publish.sz_uploaded_bytes,string"`
+	UploadedCatalogBytes int64  `json:"Publish.sz_uploaded_catalog_bytes,string"`
+	StartTime            string `json:"start_time"`
+}
+
+// TODO: concurrent access
+type StatisticsMgr struct {
+	leaseCounters map[string]Counters
+}
```

Comment:
This is not thread-safe, isn't it?

However there may be multiple worker working at the same time. It is never going to be a problem, but maybe is still worth to wrap everything in either a mutex, RWLock or use a syncMap.

Reply:
Figuring out if the `lease path` includes the name of the repository...

Reply:
Yes, the `lease_path` contains also the name of the repo, so it will not create conflicts.

Reply:
I added an RWLock. I'm not 100 % sure that distinct `lease_path`s will make it thread-safe on its own.

Reply:
No, it won't make thread safe by itself. 

I was just making sure that two repos cannot overwrite their statistics if the happen to share the same path.